conans/client/api/conan_api.py

```diff
@@ -92,26 +93,36 @@ def load_remotes(self, remote_name=None, update=False, check_updates=False):
 
 
 def api_method(f):
+    """Useful decorator for ConanAPI methods management"""
+    @functools.wraps(f)
     def wrapper(api, *args, **kwargs):
-        try:  # getcwd can fail if Conan runs on an unexisting folder
+        """Adding an extra error-management layer"""
+        error = None
+        results = None
+        try:  # getcwd can fail if Conan runs on an non-existing folder
             old_curdir = os.getcwd()
         except EnvironmentError:
             old_curdir = None
         try:
             api.create_app()
             with environment_append(api.app.cache.config.env_vars):
-                return f(api, *args, **kwargs)
-        except Exception as exc:
-            msg = exception_message_safe(exc)
-            try:
-                api.out.error("{} ({})".format(str(exc.__class__.__name__), msg))
-            except BaseException:
-                pass
-            raise
+                results = f(api, *args, **kwargs)
+        except (NotFoundException, PackageNotFoundException):
+            # This exception must be caught manually due to a server inconsistency:
+            # Artifactory API returns an empty result if the recipe doesn't exist, but
+            # Conan Server returns a 404. This probably should be fixed server side,
+            # but in the meantime we must handle it here
+            # If there are no matching recipes, let's avoid raising any error
+            pass
+        except ConanConnectionError as e:
+            error = f"There was a connection problem: {str(e)}"
+        except BaseException as e:
+            error = str(e)
         finally:
             if old_curdir:
                 os.chdir(old_curdir)
-
+            # Let's return the results and any possible error
+            return results, error
```

Comment:
@memsharded @lasote let me know your opinion about this `return` change in this decorator. In my opinion, it was the most general and simplest way to manage all the possible errors. It lets all the API consumers manage any error as they really want but I still have doubts about any other better possibility to do it.